improve LanguageVariableAPITest the test to prevent breakage

### Parent Issue

_No response_

### Problem Statement

We've seen this test fail in some of the dev branches, however, it is still ok in master 
We better improve the test to prevent breakage 
```
2024-05-23T07:05:08.4143407Z [WARNING] Flakes: 
2024-05-23T07:05:08.4146579Z [WARNING] com.dotcms.languagevariable.business.LanguageVariableAPITest.findLanguageVariablesThenUnpublishTest
2024-05-23T07:05:08.4149928Z [ERROR]   Run 1: LanguageVariableAPITest.findLanguageVariablesThenUnpublishTest:393->removeAll:536->lambda$removeAll$6:539->lambda$removeAll$5:541 » DotState No live version Contentlet. Call setLive first
2024-05-23T07:05:08.4152002Z [INFO]   Run 2: PASS
2024-05-23T07:05:08.4152818Z [INFO]
```

### Steps to Reproduce

it's uncertain how this can happen. It's a rare condition in out tests.our

### Acceptance Criteria

we should be able to build at all times with no failing tests

### dotCMS Version

master

### Proposed Objective

Reliability

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/dotcms-integration/src/test/java/com/dotcms/languagevariable/business/LanguageVariableAPITest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/languagevariable/business/LanguageVariableAPITest.java
@@ -13,9 +13,10 @@ import com.dotcms.datagen.LanguageDataGen;
 import com.dotcms.datagen.LanguageVariableDataGen;
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.CacheLocator;
-import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.business.DotStateException;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
@@ -26,6 +27,7 @@ import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
 import com.dotmarketing.portlets.languagesmanager.business.LanguageCache;
 import com.dotmarketing.portlets.languagesmanager.model.Language;
 import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.model.User;
 import com.liferay.util.SystemProperties;
@@ -35,7 +37,6 @@ import java.util.List;
 import java.util.Map;
 import java.util.concurrent.Callable;
 import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.stream.Collectors;
 import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
@@ -390,6 +391,7 @@ public class LanguageVariableAPITest extends IntegrationTestBase {
     @Test
     public void findLanguageVariablesThenUnpublishTest() throws Exception {
 
+        //destroyAll();
         destroyAll();
         final LanguageCache languageCache = CacheLocator.getLanguageCache();
         languageCache.clearVariables();
@@ -525,24 +527,26 @@ public class LanguageVariableAPITest extends IntegrationTestBase {
     }
 
 
-    /**
-     * Helper method to destroy all the contentlets created for this test
-     * @throws Exception
-     */
-    public static void destroyAll() throws Exception{
-        final ContentletAPI api = APILocator.getContentletAPI();
-        final List<String> inodes = new DotConnect().setSQL(
-                        "select working_inode from "
-                                + "contentlet_version_info cvi, "
-                                + "identifier "
-                                + "where "
-                                + "identifier.id = cvi.identifier and "
-                                + "identifier.asset_subtype='persona' ")
-                .loadObjectResults().stream().map(m -> (String) m.get("Languagevariable")).collect(
-                        Collectors.toList());
-
-        final List<Contentlet> cons = api.findContentlets(inodes);
-        api.destroy(cons, APILocator.systemUser(), false);
+    public static void destroyAll() {
+        final User sysUser = APILocator.systemUser();
+        final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+        final LanguageVariableAPI languageVariableAPI = APILocator.getLanguageVariableAPI();
+        final Map<Language, List<LanguageVariable>> allVariables = languageVariableAPI.findAllVariables();
+        allVariables.entrySet().stream().flatMap(entry -> entry.getValue().stream())
+                .forEach(languageVariable -> {
+                    try {
+                        final List<Contentlet> allVersions = contentletAPI.findAllVersions(new Identifier(languageVariable.identifier()), sysUser, false);
+                        allVersions.forEach(cont -> {
+                            try {
+                                contentletAPI.destroy(cont, sysUser, false);
+                            } catch (DotDataException | DotStateException | DotSecurityException e) {
+                                Logger.error(LanguageVariableAPITest.class, e.getMessage(), e);
+                            }
+                        });
+                    } catch (DotDataException | DotStateException | DotSecurityException e) {
+                        Logger.error(LanguageVariableAPITest.class, e.getMessage(), e);
+                    }
+                });
         final LanguageCache languageCache = CacheLocator.getLanguageCache();
         languageCache.clearVariables();
     }

--- a/dotcms-integration/src/test/java/com/dotcms/languagevariable/business/LanguageVariableAPITest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/languagevariable/business/LanguageVariableAPITest.java
@@ -2,8 +2,8 @@ package com.dotcms.languagevariable.business;
 
 import static com.dotcms.contenttype.model.type.KeyValueContentType.MULTILINGUABLE_FALLBACK_KEY;
 import static com.dotcms.integrationtestutil.content.ContentUtils.createTestKeyValueContent;
-import static com.dotcms.integrationtestutil.content.ContentUtils.updateTestKeyValueContent;
 import static com.dotcms.integrationtestutil.content.ContentUtils.deleteContentlets;
+import static com.dotcms.integrationtestutil.content.ContentUtils.updateTestKeyValueContent;
 
 import com.dotcms.IntegrationTestBase;
 import com.dotcms.contenttype.model.type.BaseContentType;
@@ -13,9 +13,9 @@ import com.dotcms.datagen.LanguageDataGen;
 import com.dotcms.datagen.LanguageVariableDataGen;
 import com.dotcms.util.IntegrationTestInitService;
 import com.dotmarketing.beans.Host;
-import com.dotmarketing.beans.Identifier;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.common.db.DotConnect;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
@@ -26,7 +26,6 @@ import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
 import com.dotmarketing.portlets.languagesmanager.business.LanguageCache;
 import com.dotmarketing.portlets.languagesmanager.model.Language;
 import com.dotmarketing.util.Config;
-import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.model.User;
 import com.liferay.util.SystemProperties;
@@ -36,6 +35,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.concurrent.Callable;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
 import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
@@ -388,9 +388,9 @@ public class LanguageVariableAPITest extends IntegrationTestBase {
      * @throws DotSecurityException
      */
     @Test
-    public void findLanguageVariablesThenUnpublishTest() throws DotDataException, DotSecurityException {
+    public void findLanguageVariablesThenUnpublishTest() throws Exception {
 
-        removeAll();
+        destroyAll();
         final LanguageCache languageCache = CacheLocator.getLanguageCache();
         languageCache.clearVariables();
         final List<LanguageVariable> vars = languageCache.getVars(1);
@@ -524,31 +524,25 @@ public class LanguageVariableAPITest extends IntegrationTestBase {
         return variables.stream().anyMatch(variable -> variable.getIdentifier().equals(identifier));
     }
 
+
     /**
-     * Method to cleanup the language variables
+     * Helper method to destroy all the contentlets created for this test
+     * @throws Exception
      */
-    public static void removeAll() {
-        final User user = APILocator.systemUser();
-        final ContentletAPI contentletAPI = APILocator.getContentletAPI();
-        final LanguageVariableAPI languageVariableAPI = APILocator.getLanguageVariableAPI();
-        final Map<Language, List<LanguageVariable>> allVariables = languageVariableAPI.findAllVariables();
-        allVariables.entrySet().stream().flatMap(entry -> entry.getValue().stream())
-                .forEach(languageVariable -> {
-                    try {
-                        final List<Contentlet> allVersions = contentletAPI.findAllVersions(new Identifier(languageVariable.identifier()), user, false);
-                        allVersions.forEach(cont -> {
-                            try {
-                                contentletAPI.unpublish(cont, user, false);
-                                contentletAPI.archive(cont, user, false);
-                                contentletAPI.delete(cont, user, false);
-                            } catch (DotDataException | DotSecurityException e) {
-                                Logger.error(LanguageVariableAPITest.class, e.getMessage(), e);
-                            }
-                        });
-                    } catch (DotDataException | DotSecurityException e) {
-                        Logger.error(LanguageVariableAPITest.class, e.getMessage(), e);
-                    }
-                });
+    public static void destroyAll() throws Exception{
+        final ContentletAPI api = APILocator.getContentletAPI();
+        final List<String> inodes = new DotConnect().setSQL(
+                        "select working_inode from "
+                                + "contentlet_version_info cvi, "
+                                + "identifier "
+                                + "where "
+                                + "identifier.id = cvi.identifier and "
+                                + "identifier.asset_subtype='persona' ")
+                .loadObjectResults().stream().map(m -> (String) m.get("Languagevariable")).collect(
+                        Collectors.toList());
+
+        final List<Contentlet> cons = api.findContentlets(inodes);
+        api.destroy(cons, APILocator.systemUser(), false);
         final LanguageCache languageCache = CacheLocator.getLanguageCache();
         languageCache.clearVariables();
     }

--- a/dotcms-integration/src/test/java/com/dotcms/languagevariable/business/LanguageVariableAPITest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/languagevariable/business/LanguageVariableAPITest.java
@@ -390,8 +390,7 @@ public class LanguageVariableAPITest extends IntegrationTestBase {
      */
     @Test
     public void findLanguageVariablesThenUnpublishTest() throws Exception {
-
-        //destroyAll();
+        
         destroyAll();
         final LanguageCache languageCache = CacheLocator.getLanguageCache();
         languageCache.clearVariables();

--- a/dotcms-integration/src/test/java/com/dotcms/rest/api/v2/languages/LanguageVariablesHelperTest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/rest/api/v2/languages/LanguageVariablesHelperTest.java
@@ -1,24 +1,18 @@
 package com.dotcms.rest.api.v2.languages;
 
 
-import static com.dotcms.languagevariable.business.LanguageVariableAPITest.removeAll;
+import static com.dotcms.languagevariable.business.LanguageVariableAPITest.destroyAll;
 
 import com.dotcms.datagen.LanguageDataGen;
 import com.dotcms.datagen.LanguageVariableDataGen;
-import com.dotcms.languagevariable.business.LanguageVariable;
 import com.dotcms.languagevariable.business.LanguageVariableAPI;
 import com.dotcms.languagevariable.business.LanguageVariableAPITest;
 import com.dotcms.util.IntegrationTestInitService;
-import com.dotmarketing.beans.Identifier;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
-import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
-import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.portlets.languagesmanager.model.Language;
-import com.dotmarketing.util.Logger;
 import com.fasterxml.jackson.core.JsonProcessingException;
-import com.liferay.portal.model.User;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
@@ -48,11 +42,11 @@ public class LanguageVariablesHelperTest {
      */
     @Test
     public void paginationTestVerifyKeyOrder()
-            throws DotDataException, DotSecurityException, JsonProcessingException {
+            throws Exception {
 
         final LanguageVariableAPI languageVariableAPI = APILocator.getLanguageVariableAPI();
 
-        removeAll();
+        destroyAll();
         Assert.assertEquals(0, languageVariableAPI.countVariablesByKey());
 
         final List<Language> languages = List.of(

